### DISCUSSION TITLE
Pasting a transparent image over a canvas .

### DISCUSSION THREAD

**Discussion by httpanand** [python-pillow/Pillow#5855] [Q&A] [answered]
I need to paste a transparent image over a new image canvas made with pillow . But it automatically adds a black background in the output image . Can anyone help me to fix  this ? Below is my code . 

```py
from PIL import Image, ImageDraw, ImageFont , ImageOps , ImageColor


image = Image.new('RGB', (250,400), 'red')
image = ImageOps.expand(image,border=4,fill='black')

icon = Image.open('icon.png')
image.paste(icon ,(150,295))


image.save("output.png")
```

**Comment by radarhere** [comment, marked as answer]
In the interest of giving you a quick answer, I suggest trying this -


```py
image.paste(icon ,(150,295), icon)
```

**Reply by radarhere** [reply]
This is making use of the `mask` parameter - https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.paste

**Reply by httpanand** [reply]
I knew `alpha_composite()` . But that doesn't work for me .. When i call that func , the whole canvas disappears !

**Comment by httpanand** [comment]
OMG Thanks <3 . @radarhere
